pkg/backup/item_collector.go

```diff
@@ -289,16 +291,43 @@ func (r *itemCollector) getResourceItems(log logrus.FieldLogger, gv schema.Group
 		}
 
 		log.Info("Listing items")
-		unstructuredList, err := resourceClient.List(metav1.ListOptions{LabelSelector: labelSelector})
+		var unstructuredList *unstructured.UnstructuredList
+		var unstructuredItems []unstructured.Unstructured
+		var continueValue string
+
+		if r.listLimit == 0 {
+			// If limit is the special value 0, do not use a limit and request all items at once
+			unstructuredList, err = resourceClient.List(metav1.ListOptions{LabelSelector: labelSelector})
+		} else {
+			unstructuredList, err = resourceClient.List(metav1.ListOptions{LabelSelector: labelSelector, Limit: int64(r.listLimit)})
+		}
 		if err != nil {
 			log.WithError(errors.WithStack(err)).Error("Error listing items")
 			continue
 		}
+		unstructuredItems = append(unstructuredItems, unstructuredList.Items...)
+
+		if r.listLimit > 0 {
+			// If using a limit, continue requesting chunks of the list
+			continueValue = unstructuredList.GetContinue()
+			for continueValue != "" {
+				unstructuredList, err := resourceClient.List(metav1.ListOptions{LabelSelector: labelSelector, Limit: int64(r.listLimit), Continue: continueValue})
+				if statusError, isStatus := err.(*k8serrors.StatusError); isStatus && statusError.Status().Reason == metav1.StatusReasonExpired {
+					log.Warnf("Got ResourceExpired error while listing items. Resource list may be incomplete due to modifications while listing items")
+				} else if err != nil {
+					log.WithError(errors.WithStack(err)).Error("Error listing items")
+					break
+				}
+				unstructuredItems = append(unstructuredItems, unstructuredList.Items...)
+				continueValue = unstructuredList.GetContinue()
```

Comment:
as mentioned in the other comments, would it make sense to retry implementing this with the client-go Pager type?

Reply:
I did that this morning! I've got a patch using the pager locally I'm testing out today. If my functional tests look good expect a push later today.